expense-notifications-functions/lib/index.js

```
@@ -0,0 +1,404 @@
+                return 0; // Avoid division by zero
+            return (amount * ((splitData)[member] || 0)) / totalShares;
+        }
+        default:
+            return 0;
+    }
+}
+// Cloud Function to handle add transaction and notify users
+exports.onTransactionCreated = (0, firestore_2.onDocumentCreated)({ document: 'groups/{groupId}/transactions/{transactionId}' }, async (event) => {
+    var _a;
+    try {
+        const transactionData = (_a = event.data) === null || _a === void 0 ? void 0 : _a.data();
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)